### DISCUSSION TITLE
Setlist Song Display not following Arrangement Markers

### DISCUSSION THREAD

**Discussion by GTraux** [KBLiveSolutions/open.control#15] [Troubleshooting]
Hi! Don't know if I am doing something wrong, but the display on open.control is not following along the (SONG#) in the Arrangement View. I thought it would be updating along each time the transport passed into another Marker with (SONG#). Is my premise wrong?

Just for clarification, the SONG:LOOPER:MODE is seeing all the Markers with (SONG#) and grouping them as expected into a song. 

My ideal objective is to have 3 Arrangement Markers per track on my liveset: one for the start of the track itself and the other two as a looping region where we usually loop material for each track and where I would use the Arrangement Looper function to go in and out.

The perfect scenario would be open.control follow such changes and allow me to see and control the liveset from my side of the desk.

Hope this makes sense.

Thanks!

**Comment by KBLiveSolutions** [comment]
Hi

Yes it makes perfectly sense, and as of now, setlists in open·control are not fully compatible with SONG:LOOPER:MODE.
I plan to make it compatible soon, so that if there are several Markers/Scenes that have the same (SONG#), they will appear as #.1, #.2, #.3 and so on.
Right now, if several Markers/Scenes have the same (SONG#), only the last one will be added to the Setlist.

I hope this is clear, I will fix that asap.

**Reply by GTraux** [reply]
Yep! Good to know, Pierre! This hybrid use of Session/Arrangement using the display as a remote controller/visual monitor is insane for the stage. The use I pretend is to have the markers as cues of gestures needed (e.g.: loop this, filter that, automate this, fade to grey that, etc.), like a dynamic checklist to aid remember stuff over 2h-3h of live music, without staring into the computer or a sheet of paper on the floor. Hope this can work!

I've taken out the other SONGn in the Markers for now to avoid any compatibility issue. I was indeed having some glitches while using the "Loop to the Next Marker". It was working 90% of the time, but would hiccup around some markers. I'll check now if that still persists and came back to report.

**Comment by KBLiveSolutions** [comment]
The updated Remote script should partly solve this issue too.

https://github.com/KBLiveSolutions/open.control-remote_script/archive/refs/heads/custom-actions.zip

If several Scenes/Markers have the same (SONG#), they will appear as #.1, #.2 etc.
For now it's not possible to launch them, only select, and I haven't checked how it behaves with Markers.

**Reply by KBLiveSolutions** [reply]
I have just changed the script, you are now able to launch Setlist songs in Session and Arrangement view.
The only thing that is missing, is that the display won't be updated when a Marker is passed.
This will be fixed later.

**Reply by GTraux** [reply]
This is so nice!!! Thanks for doing this! Rehearsing the liveset this exact moment. Will do the script update later and test.

**Reply by KBLiveSolutions** [reply]
Another update to the script : the display for Setlist Song will follow the Arrangement, so the display is always showing the current Setlist position.
Download and install the updated Remote Script :
https://github.com/KBLiveSolutions/open.control-remote_script/archive/refs/heads/custom-actions.zip
